examples/gpio/config.c

```diff
@@ -0,0 +1,248 @@
+
+   if (argc >= 2) {
+        if (XSTRNCMP(argv[1], "-?", 2) == 0 ||
+            XSTRNCMP(argv[1], "-h", 2) == 0 ||
+            XSTRNCMP(argv[1], "--help", 6) == 0) {
+            usage();
+            return 0;
+        }
+        if (argc == 3) {
+            gpioMode = atoi(argv[2]);
+            if (gpioMode > 5) {
+                printf("GPIO mode is out of range (0-5)\n");
+                usage();
+                goto exit_badargs;
+            }
+            /* Check if mode is for GPIO Input to perform extra config step */
+            if (gpioMode > 0 && gpioMode < 4) {
+                gpioInput = 1;
+            }
+            gpioMode += TPM_GPIO_MODE_STANDARD;
+            /* Preparing to process next argument */
+            argc--;
+        }
+        if (argc == 2) {
+            gpioNum = atoi(argv[1]);
+            if (gpioNum < 0 || gpioNum > 3) {
+                printf("GPIO is out of range (0-3)\n");
+                usage();
+                goto exit_badargs;
+            }
+            /* Increment NV index used for this GPIO */
+            nvIndex += gpioNum;
+            /* Index GPIO name */
+            gpioNum += TPM_GPIO_A;
+            /* all arguments processed */
+        }
+    }
+    else if (argc == 1) {
+        /* Default behavior, without arguments: GPIO 0 as output */
+        gpioMode = TPM_GPIO_MODE_PUSHPULL;
+        gpioNum = TPM_GPIO_A;
+    }
+    else {
+        printf("Incorrect arguments\n");
+        usage();
+        goto exit_badargs;
+    }
+
+    printf("GPIO num is: %d\n", gpioNum);
+    printf("GPIO mode is: %d\n", gpioMode);
+
+    /* Sanity check TPM_GPIO_B can be used only as input */
+    if (gpioNum == TPM_GPIO_B &&
+        (gpioMode == TPM_GPIO_MODE_PUSHPULL ||
+        gpioMode == TPM_GPIO_MODE_OPENDRAIN)) {
+        printf("Warning: TPM_GPIO_B can be used only as an input.\n");
+        usage();
+        return 0;
+    }
+
+    printf("Demo how to use extra GPIO on a TPM 2.0 modules\n");
+    rc = wolfTPM2_Init(&dev, TPM2_IoCb, userCtx);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("wolfTPM2_Init failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+        goto exit;
+    }
+    printf("wolfTPM2_Init: success\n");
+
+    /* Get TPM capabilities, to discover the TPM vendor */
+    rc = wolfTPM2_GetCapabilities(&dev, &caps);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("wolfTPM2_GetCapabilities failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+    }
+
+    /* Enable TPM2_GPIO_Config command */
+    if (caps.mfg != TPM_MFG_STM) {
+        printf("Extra GPIO is supported only on ST33 TPM 2.0 modules\n");
+        goto exit;
+    }
+
+#ifdef DEBUG_WOLFTPM
+    printf("Trying to remove NV index 0x%8.8X used for GPIO\n", nvIndex);
+#endif
+    /* Make sure NV Index for this GPIO is cleared before use
+     * This way we make sure a new GPIO config can be set
+     */
+    rc = wolfTPM2_NVDelete(&dev, TPM_RH_OWNER, nvIndex);
+#ifdef DEBUG_WOLFTPM
+    if (rc == TPM_RC_SUCCESS) {
```

Comment:
Please do not report an error if the NV does not exist. scan-build will report warning `rc` set but not read. Please keep some logic in here for non-debug.